plugins/suggested-blocks/README.md

```diff
@@ -21,24 +21,40 @@ import {toolboxCategories} from '@blockly/dev-tools';
 import * as SuggestedBlocks from '@blockly/suggested-blocks';
 
 function getNewCategories() {
-  const mostUsedCategory = '<category name="Frequently Used" custom="MOST_USED"></category>';
-  const recentlyUsedCategory = '<category name="Recently Used" custom="RECENTLY_USED"></category>';
-
-  // This type of insertion works in a pinch, but you would more so want to add
-  // the new categories wherever the rest of the categories are defined in the
-  // project (e.g. HTML, or a dedicated file for all the categories)
-  const indexToInsert = toolboxCategories.indexOf('</xml>');
-  // Insert the new categories into the existing toolbox XML string
-  return toolboxCategories.slice(0, indexToInsert) + mostUsedCategory + recentlyUsedCategory + toolboxCategories.slice(indexToInsert);
+  const mostUsedCategory = {
+    kind: 'category',
+    name: 'Frequently Used',
+    custom: 'MOST_USED'
+  };
+  const recentlyUsedCategory = {
+    kind: 'category',
+    name: 'Recently Used',
+    custom: 'RECENTLY_USED'
+  };
+  
+  // Insert the new categories into the existing toolboxCategories array
+  return [
+    ...toolboxCategories.slice(0, -1),
```

Comment:
The `toolboxCategories` object is [here](https://github.com/google/blockly-samples/blob/master/plugins/dev-tools/src/toolboxCategories.js). It's a json object, not an array. You need to modify the `toolboxCategories.contents` array. Since we're just adding to the end of the array, you can use `push` or destructuring but you don't need to use `slice`.

Reply:
Sorry, after looking at this again, we actually don't need to use the `toolboxCategories` from `@blockly/dev-tools` at all. This code is in the README where we are showing developers how to use this plugin. The `toolboxCategories` object is provided as a convenience when developing plugins, but it's not intended for use outside of plugins, so we shouldn't include it in the README.

So what we should do is show a snippet of a self-contained toolbox. Since it's just an example, the toolbox can be pretty bare-bones apart from our two special categories we're showing from this plugin. Here's an example, which you can use or come up with your own.

```
const toolbox = {
  'kind': 'categories',
  'contents': [
    // Your own categories and blocks here...
    {
      'kind': 'category',
      'name': 'Frequently Used',
      'custom': 'MOST_USED',
    },
    {
      'kind': 'category',
      'name': 'Recently Used',
      'custom': 'RECENTLY_USED',
    },
  ],
};
```

This toolbox only has two categories, which isn't super useful, but a developer can learn how to specify toolboxes in general from other documentation.